plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/aggregation/ImplementSum.java

```diff
@@ -81,8 +84,9 @@ else if (aggregateFunction.getOutputType() instanceof DecimalType) {
         }
 
         ParameterizedExpression rewrittenArgument = context.rewriteExpression(argument).orElseThrow();
+        String function = aggregateFunction.isDistinct() ? "sum(DISTINCT %s)" : "sum(%s)";
```

Comment:
As I understand the pushdown was working now, but it was incorrect as nobody checked `isDistinct()`. Am I right? If so then it is a bugfix.

Reply:
Previous the pattern `distinct` is equals to `false`, so the `sum(distinct)` was not pushdown.